### PR TITLE
Fix performance issues in editor components

### LINKED ISSUE #4885: 0.21.0 Desktop | Very slow editing markdown and code cell blocks
**Application or Package Used**
nteract desktop 0.21.0 | macOS 10.15.2
 - upgraded from nteract desktop 0.15.0 to 0.21.0 using dmg file release

**Describe the bug**
Editing cells is very slow and sluggish.

For example: When I type in '3333333' just fast, it takes about four seconds to render it.

**Expected behavior**

- Rendering while editing should be fluid.

**Desktop (please complete the following information):**

macOS 10.15.2 & macOS 10.15.3

### PR DESCRIPTION
**Pull request by captainsafia**
Closes https://github.com/nteract/nteract/issues/4885.

For more information on the changes in `packages/editor/src/index.tsx` and `packages/stateful-components/src/inputs/editor.tsx` please see [this comment](https://github.com/nteract/nteract/issues/4885#issuecomment-581191883).

The changes in `packages/stateful-components/src/cells/cell.tsx` were made to resolve an issue I noticed where the `Cell` parent component was re-rendered every time anything in the cell changed since the `Cell` component takes the entire ImmutableCell object as a prop.

### CODE REVIEW

File: packages/stateful-components/src/inputs/editor.tsx [hunk 1]
```diff
@@ -63,7 +70,14 @@ export class Editor extends React.Component<ComponentProps & StateProps> {
 
     // Render the output component that handles this output type
     return React.cloneElement(chosenOne, {
-      ...this.props,
```
Comment (eimfach) [inline packages/stateful-components/src/inputs/editor.tsx:66]: Is it necessary to copy all this props ?

File: packages/stateful-components/src/cells/cell.tsx [hunk 2]
```diff
@@ -15,7 +15,13 @@ interface StateProps {
   selected: boolean;
 }
 
-export class Cell extends React.PureComponent<ComponentProps & StateProps> {
+type Props = ComponentProps & StateProps;
+
+export class Cell extends React.Component<ComponentProps & StateProps> {
```
Comment (eimfach) [inline packages/stateful-components/src/cells/cell.tsx]: I am not really experienced with TypeScript, but wouldn't placing the new type alias into the class signature be better ?
`export class Cell extends React.Component<Props>`

File: packages/stateful-components/src/inputs/editor.tsx (hunk 1 above)
Comment (captainsafia) [reply packages/stateful-components/src/inputs/editor.tsx:66]: Yes, they are the props that are commonly required by editor components. Before, we were passing all the props. Now we are passing all of them, except `children`.

File: packages/stateful-components/src/cells/cell.tsx (hunk 2 above)
Comment (captainsafia) [reply packages/stateful-components/src/cells/cell.tsx]: Good catch! I will update this.

File: packages/stateful-components/src/inputs/editor.tsx (hunk 1 above)
Comment (eimfach) [reply packages/stateful-components/src/inputs/editor.tsx:66]: Okay, and what if only `value` changed ? Wouldn't it be copying all those other props repeatedly on every keystroke ? Just a thought ...

Comment (eimfach) [reply packages/stateful-components/src/inputs/editor.tsx:66]: I understand that this is convenient to do like this ...

Comment (captainsafia) [reply packages/stateful-components/src/inputs/editor.tsx:66]: Hmmm. Yes, we should probably add a stronger `shouldComponentUpdate` check here as well. I'll have to try this out later and validate how we can optimize this.

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
Thanks for the review @eimfach! I'm going to go ahead with merging this to continue iterations in follow on PRs.

**Pull request merged**